model/LTIDeliveryTool.php

```diff
@@ -78,7 +78,10 @@ public function getFinishUrl(LtiMessage $ltiMessage, $deliveryExecution = null)
                 $urlParts['query'] = '';
             }
             parse_str($urlParts['query'], $params);
-            $params = array_merge($params, $ltiMessage->getUrlParams());
+            if ($ltiMessage) {
+                $params = array_merge($params, $ltiMessage->getUrlParams());
+            }
+            $params['deliveryExecution'] = $deliveryExecution->getIdentifier();
```

Comment:
what if deliveryExecution is null ?

Reply:
We have condition on line 75 to check if it is `null`